### PR TITLE
Zebras: time length fixes.

### PR DESCRIPTION
**Pull request by jaheba**
* Rework `split_frame` length calculation.
* Fix missing handling of tdims.
* Add consistency checks for `SplitFrame`
* Allow integer index access on TimeSeries

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

File: src/gluonts/zebras/_split_frame.py [hunk 1]
```diff
@@ -42,6 +43,14 @@ class SplitFrame:
     default_tdim: int = -1
     _pad: Pad = Pad()
 
+    def __post_init__(self):
+        for column in itertools.chain(self._past, self._future):
+            self.tdims.setdefault(column, self.default_tdim)
+
+        # this triggers checks for past_length and future_length
+        self.past
+        self.future
```
Comment (abdulfatir) [inline src/gluonts/zebras/_split_frame.py]: I don't know if this is how it's generally done. Would something like this be more readable:
```py
_, _ = self.past, self.future
```

File: src/gluonts/zebras/_split_frame.py [hunk 2]
```diff
@@ -357,47 +366,55 @@ def split_frame(
...
+                "Cannot determine past and future length if only "
+                "`full` is provided."
+            )
+    # No data and no lengths are passed
+    elif maybe.or_(past_length, future_length) is None:
+        past_length = 0
+        future_length = 0
+        full_length = 0
+    else:
+        full_length = None
```
Comment (abdulfatir) [inline src/gluonts/zebras/_split_frame.py]: What is this case?

File: src/gluonts/zebras/_split_frame.py (hunk 1 above)
Comment (jaheba) [reply src/gluonts/zebras/_split_frame.py]: Maybe, I'm not sure. But let's do this.

File: src/gluonts/zebras/_split_frame.py (hunk 2 above)
Comment (jaheba) [reply src/gluonts/zebras/_split_frame.py]: Let me add a comment.

**Review by abdulfatir — APPROVED** [review]

**Pull request merged**